src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java

```diff
@@ -83,6 +85,11 @@ public abstract class DefaultStepContext extends StepContext {
             value = castOrNull(key, getExecution());
         } else if (FlowNode.class.isAssignableFrom(key)) {
             value = castOrNull(key, getNode());
+        } else if (StepDescriptor.class.isAssignableFrom(key)) {
+            var stepNode = castOrNull(StepNode.class, getNode());
+            if (stepNode != null) {
+                value = castOrNull(key, stepNode.getDescriptor());
```

Comment:
Sorry, my mistake, in fact https://javadoc.jenkins.io/plugin/workflow-api/org/jenkinsci/plugins/workflow/graph/StepNode.html#getDescriptor() is `@CheckForNull` though I guess it does not matter since https://github.com/jenkinsci/workflow-support-plugin/blob/a0887e7be9317a204609ba64ec8fdf5e9b72a37b/src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java#L163 will be false in that case anyway.

Reply:
right, works, tested `castOrNull(key, null)` returns null.

so I think additional null check is not required in this case.